### PR TITLE
Staging Sites: Hide staging sites module on Pro Plan

### PR DESCRIPTION
**Pull request by katinthehatsite**
* Closes https://github.com/Automattic/dotcom-forge/issues/2019

## Proposed Changes

* This PR hides the staging sites card on the sites that have Pro Plan. 

## Testing Instructions

* Make sure you have a WordPress.com site with a Pro Plan (note: it is a legacy plan, you would need to add it to a site from Store Admin)
* Instal any plugin on that site to make it Atomic
* Navigate to `Settings > Hosting Configuration` : /hosting-config/{site}
* Check that the **Staging Sites** module is not present:

Screenshot 2023-03-30 at 3 16 24 PM

* Go to an Atomic site with a Business plan
* Navigate to `Settings > Hosting Configuration` : /hosting-config/{site}
* Check that the Staging Site module is present:

Screenshot 2023-03-30 at 3 18 59 PM

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/hosting/main.js [hunk 1]
```diff
@@ -191,7 +192,7 @@ class Hosting extends Component {
 							<Column type="main" className="hosting__main-layout-col">
 								<SFTPCard disabled={ isDisabled } />
 								<PhpMyAdminCard disabled={ isDisabled } />
-								{ isStagingSiteEnabled && ! isWpcomStagingSite && (
+								{ isStagingSiteEnabled && ! isWpcomStagingSite && ! isProPlan( sitePlanSlug ) && (
```
Comment (wojtekn) [inline client/my-sites/hosting/main.js]: This should work for the hiding staging UI for the Pro plan. What do you think about making this check more generic, to operate on the feature and not on the plan level? That way, we could fix other uncovered cases in which hosting features should be enabled but staging sites shouldn't.

An example of a similar approach could be the `hasSftpFeature` check that exists in the same file.

Comment (katinthehatsite) [reply client/my-sites/hosting/main.js]: Thanks for the pointer! I implemented the feature check instead of the plan check - let me know what you think.

File: client/my-sites/hosting/main.js [hunk 2]
```diff
@@ -177,6 +180,7 @@ class Hosting extends Component {
 			const isGithubIntegrationEnabled = isAutomatticTeamMember( teams );
 			const WrapperComponent = isDisabled || isTransferring ? FeatureExample : Fragment;
 			const isStagingSiteEnabled = isEnabled( 'yolo/staging-sites-i1' );
+			const { hasStagingSites } = this.props;
```
Comment (wojtekn) [inline client/my-sites/hosting/main.js]: Would it be better to destruct it upper in the stack, as a part of `render()` where we destruct other props?

**Review by wojtekn — APPROVED** [review]
I added one comment, but looks like good to go. Works as expected for Pro plan.

File: client/my-sites/hosting/main.js [hunk 3]
```diff
@@ -238,6 +242,7 @@ export default connect(
 	( state ) => {
 		const siteId = getSelectedSiteId( state );
 		const hasSftpFeature = siteHasFeature( state, siteId, FEATURE_SFTP );
+		const hasStagingSites = siteHasFeature( state, siteId, FEATURE_SITE_STAGING_SITES );
```
Comment (sejas) [inline client/my-sites/hosting/main.js]: Can we make it more explicit that the site has the staging sites "feature" ?

```suggestion
		const hasStagingSitesFeature = siteHasFeature( state, siteId, FEATURE_SITE_STAGING_SITES );
```

**Review by sejas — APPROVED** [review]
@katinthehatsite, Well done!

It works as expected. 👌 

I added a small name suggestion.

You'll need to resolve a git conflict, happy to help if needed.

We'll need to deploy [ETK](https://github.com/Automattic/wp-calypso/pull/75127#issuecomment-1493988918) too. Let's do that together.

File: packages/data-stores/src/plans/types.ts [hunk 4]
```diff
@@ -61,7 +61,7 @@ export interface PricedAPIPlan {
 	product_slug: StorePlanSlug;
 	product_name_short: string;
 	product_type?: string;
-	bill_period: -1 | ( typeof PERIOD_LIST )[ number ];
+	bill_period: -1 | typeof PERIOD_LIST[ number ];
```
Comment (sejas) [inline packages/data-stores/src/plans/types.ts]: Let's undo the changes in this file to fix the code style tests.

You can override the content of this file from trunk:
`git checkout origin/trunk -- packages/data-stores/src/plans/types.ts`

**Comment by sejas** [comment]
We deployed ETK that only bumped the version r264582-wpcom
